GitHub.Authentication/Authentication.cs

```diff
@@ -57,22 +57,16 @@ public class Authentication : BaseAuthentication, IAuthentication
             AcquireAuthenticationCodeDelegate acquireAuthenticationCodeCallback,
             AuthenticationResultDelegate authenticationResultCallback)
         {
-            if (tokenScope == null)
-                throw new ArgumentNullException("tokenScope", "The parameter `tokenScope` is null or invalid.");
-            if (personalAccessTokenStore == null)
-                throw new ArgumentNullException("personalAccessTokenStore", "The parameter `personalAccessTokenStore` is null or invalid.");
-            if (acquireCredentialsCallback == null)
-                throw new ArgumentNullException("acquireCredentialsCallback", "The parameter `acquireCredentialsCallback` is null or invalid.");
-            if (acquireAuthenticationCodeCallback == null)
-                throw new ArgumentNullException("acquireAuthenticationCodeCallback", "The parameter `acquireAuthenticationCodeCallback` is null or invalid.");
-
-            TokenScope = tokenScope;
+            TokenScope = tokenScope
+                ?? throw new ArgumentNullException("tokenScope", "The parameter `tokenScope` is null or invalid.");
```

Comment:
trendy... I seriously wonder how long this will be considered "the way to do it"? Maybe it'll stick, but to a non-C# developer this must ready like insanity. :smirk:

Reply:
Once it's in the GCM, everybody will  do this. And it won't read like insanity to F# developers. :smile: